packages/flutter_tools/lib/src/web/compile.dart

```diff
@@ -28,10 +28,13 @@ Future<void> buildWeb(
   bool nativeNullAssertions,
   String? baseHref,
   String? dart2jsOptimization,
+  [String? outputDirectoryPath,]
 ) async {
   final bool hasWebPlugins = (await findPlugins(flutterProject))
     .any((Plugin p) => p.platforms.containsKey(WebPlugin.kConfigKey));
-  final Directory outputDirectory = globals.fs.directory(getWebBuildDirectory());
+  final Directory outputDirectory = outputDirectoryPath == null
+      ? globals.fs.directory(getWebBuildDirectory())
+      : globals.fs.directory(outputDirectoryPath);
```

Comment:
Will we recursively create the directory if some of its parents don't already exist?

Reply:
Yes we will, the `recursive` parameter has been marked as true in the line below

`outputDirectory.createSync(recursive: true);`